packages/core/src/state-management/state-manager.ts

```diff
@@ -340,14 +342,69 @@ export class StateManager {
     return this._processAnchorResponse(state$, anchorStatus$)
   }
 
-  private async _saveAnchorRequestForState(state$: RunningState): Promise<void> {
+  private async _buildAnchorRequestCARFile(streamId: StreamID, tip: CID): Promise<CAR> {
+    const carFactory = new CARFactory()
+    carFactory.codecs.add(DAG_JOSE)
+    const car = carFactory.build()
+
+    // Root block
+    const timestampISO = new Date().toISOString()
+    car.put(
+      {
+        timestamp: timestampISO,
+        streamId: streamId.bytes,
+        tip: tip.bytes,
+      },
+      { isRoot: true }
+    )
+
+    // Genesis block
+    const genesisCid = streamId.cid
+    const genesisBlock = await this.dispatcher._ipfs.block.get(genesisCid)
+    const genesisCarBlock = new CarBlock(genesisCid, genesisBlock)
+    car.blocks.put(genesisCarBlock)
```

Comment:
You do this lots of times in this method. Why not create a helper function? 

```ts

```



```suggestion
    const cidToBlock = cid => new CarBlock(cid, await this.dispatcher._ipfs.block.get(cid))

    car.blocks.put(cidToBlock(streamId.cid))
```